grpc/doc/swdesign/README.md

```diff
@@ -305,6 +311,135 @@ Needs:
+
+Needs:
+- impl
+- itest
+
+#### Deactivate mTLS on gRPC server no certificates and no key provided upon the start
+`swdd~grpc-server-deactivate-mtls-when-no-certificates-and-no-key-provided-upon-start~1`
+
+Status: approved
+
+When no root certificate, no server certificate and no server key are provided upon start of the gRPC server, the gRPC server shall establish a gRPC communication between the involved parties without mTLS.
+
+Rationale:
+To avoid complexity, coming with mTLS configuration e.g. certificates generation and management, during development phase, mTLS can be activated on demand.
+
+Needs:
+- impl
+- itest
+
+#### Activate mTLS on gRPC agent connection when certificates and key are provided upon the start
+`swdd~grpc-agent-activate-mtls-when-certificates-and-key-provided-upon-start~1`
+
+Status: approved
+
+When the root certificate, the agent certificate and the agent key is provided upon start of the gRPC agent connection, the gRPC agent client shall use the provided certificates and the key to activate mTLS for the gRPC communication between the involved parties.
+
+Needs:
+- impl
+- itest
+
+#### Deactivate mTLS on gRPC agent connection when no certificates and no key provided upon the start
+`swdd~grpc-agent-deactivate-mtls-when-no-certificates-and-no-key-provided-upon-start~1`
+
+Status: approved
+
+When no root certificate, no agent certificate and no agent key are provided upon start of the gRPC agent connection, the gRPC agent client shall establish a gRPC communication between the involved parties without mTLS.
+
+Rationale:
+To avoid complexity, coming with mTLS configuration e.g. certificates generation and management, during development phase, mTLS can be activated on demand.
+
+Needs:
+- impl
+- itest
+
+#### Activate mTLS on gRPC cli connection when certificates and key are provided upon the start
+`swdd~grpc-cli-activate-mtls-when-certificates-and-key-provided-upon-start~1`
+
+Status: approved
+
+When the root certificate, the cli certificate and the cli key is provided upon start of the gRPC cli connection, the gRPC cli client shall use the provided certificates and the key to activate mTLS for the gRPC communication between the involved parties.
+
+Needs:
+- impl
+- itest
+
+#### Deactivate mTLS on gRPC cli connection when no certificates and no key provided upon the start
+`swdd~grpc-cli-deactivate-mtls-when-no-certificates-and-no-key-provided-upon-start~1`
+
+Status: approved
+
+When the root certificate, the cli certificate and the cli key is provided upon start of the gRPC cli connection, the gRPC cli client shall use the provided certificates and the key to activate mTLS for the gRPC communication between the involved parties.
+
+Rationale:
+To avoid complexity, coming with mTLS configuration e.g. certificates generation and management, during development phase, mTLS can be activated on demand.
+
+Needs:
+- impl
+- itest
```

Comment:
The two requirements have the same req text. In addition, I am not sure if we need the second requirement as a negative req.

Reply:
ok, I adapted it.

Reply:
The req looks good to me. But the other requirements about deactivating mTLS (server, agent) above it use the "and" in the preconditions. I think we need to adapt them also to "or" and maybe to use the "if" signaling unwanted behavior?

Reply:
Will be fixed with the new comment.